contentcuration/contentcuration/frontend/channelEdit/utils.js

```diff
@@ -150,3 +156,141 @@ export function assessmentItemKey(assessmentItem) {
     assessment_id: assessmentItem.assessment_id,
   };
 }
+
+/**
+ * Converts a value in seconds to a human-readable format.
+ * If the value is greater than or equal to one hour, the format will be hh:mm:ss.
+ * If the value is less than one hour, the format will be mm:ss.
+ *
+ * @param {Number} seconds - The value in seconds to be converted.
+ * @returns {String} The value in human-readable format.
+ */
+export function secondsToHms(seconds) {
+  const hours = Math.floor(seconds / 3600);
+  const minutes = Math.floor((seconds - hours * 3600) / 60);
+  const remainingSeconds = seconds - hours * 3600 - minutes * 60;
+
+  let hms = '';
+  if (hours > 0) {
+    hms += hours.toString().padStart(2, '0') + ':';
+  }
+  hms += minutes.toString().padStart(2, '0') + ':' + remainingSeconds.toString().padStart(2, '0');
+  return hms;
+}
+
+/**
+ * Gathers data from a content node related to its completion.
+ *
+ * @param {Object} node
+ * @returns {Object} { completionModel, completionThreshold, masteryModel,
+ *                     suggestedDurationType, suggestedDuration}
+ */
+export function getCompletionDataFromNode(node) {
+  if (!node) {
+    return;
+  }
+
+  const { model, threshold } = node.extra_fields?.options?.completion_criteria || {};
+  const masteryModel = threshold?.mastery_model;
+  const suggestedDurationType = node.extra_fields?.suggested_duration_type;
+  const suggestedDuration = node.suggested_duration;
+
+  return {
+    completionModel: model,
+    completionThreshold: threshold,
+    masteryModel,
+    suggestedDurationType,
+    suggestedDuration,
+  };
+}
+
+/**
+ * @param {Object} node
+ * @returns {Boolean, undefined}
+ */
+export function isLongActivity(node) {
+  if (!node) {
+    return;
+  }
+  const { completionModel, suggestedDuration } = getCompletionDataFromNode(node);
+  return (
+    completionModel === CompletionCriteriaModels.APPROX_TIME &&
+    suggestedDuration > SHORT_LONG_ACTIVITY_MIDPOINT
+  );
+}
+
+/**
+ * Determines completion and duration labels from completion
+ * criteria and related of a content node.
+ *
+ * @param {Object} node
+ * @returns {Object} { completion, duration }
+ */
+export function getCompletionCriteriaLabels(node) {
+  if (!node) {
+    return;
+  }
+
+  const {
+    completionModel,
+    completionThreshold,
+    masteryModel,
+    suggestedDuration,
+  } = getCompletionDataFromNode(node);
+
+  const labels = {
+    completion: '-',
+    duration: '-',
+  };
+
+  switch (completionModel) {
+    case CompletionCriteriaModels.REFERENCE:
+      labels.completion = metadataStrings.$tr('reference');
+      break;
+
+    case CompletionCriteriaModels.TIME:
+      labels.completion = metadataStrings.$tr('completeDuration');
+      if (suggestedDuration) {
+        labels.duration = secondsToHms(suggestedDuration);
```

Comment:
I'm not entirely sure if I should be using the suggested duration (`node.suggested_duration`) or rather threshold value (`node.extra_fields.options.completion_criteria.threshold`). Often, these two values are the same. Advice?

Reply:
I _think_ this was intentional - the non-changing of the suggested_duration is to retain a suggested duration value, even when the threshold no longer records any sort of duration value.

With that in mind, I think sticking with the suggested_duration makes the most sense, as it retains a value that is displayed to the user in Kolibri, even when the completion criteria is changed to one that does not store any duration value.

Reply:
Thank you, both.

In the future, could an option to set the suggested duration separately from the completion criteria on the edit modal bring some more clarity to this, or was it discussed already? It's a bit confusing, but primarily it seems that with this approach, there'd be no other way to update the suggested duration of a resource than to change the completion criteria to one that allows duration adjustment and then switch back to the original completion criteria.